### PR TITLE
Widen pillow dependency requirement

### PR DESCRIPTION
**Pull request by aaron-siegel**
This fixes an issue where colab complains that the kernel has to be restarted after pip-installing Pixeltable.

### CODE REVIEW

**Comment by aaron-siegel** [comment]
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/160)

File: pyproject.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ exclude = [
 python = ">=3.9,<4.0"
 numpy = ">=1.26"
 pandas = ">=2.0,<3.0"
-pillow = ">=10.0"
+pillow = ">=9.3.0"
```
Comment (mkornacker) [inline pyproject.toml:22]: We can't do that. The signature of at least one function changed between 9 and 10.

Comment (aaron-siegel) [reply pyproject.toml:22]: I ran the unit tests and they all passed.

We HAVE to do this or it causes unacceptable user friction in colab. Tell me which function changed and I'll look at it.

File: pixeltable/tests/functions/test_pil.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from ..utils import get_image_files
+
+
+class TestPil:
+
+    def to_img(self, x: Any) -> PIL.Image:
+        return x
+
+    def test_pil(self, img_tbl: Table) -> None:
+        #mask_img = next(f for f in get_image_files() if f.endswith('n03888257_1389.JPEG'))
```
Comment (mkornacker) [inline pixeltable/tests/functions/test_pil.py]: remove

File: pixeltable/tests/functions/test_pil.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+from pixeltable.functions.pil.image import *
+from ..utils import get_image_files
+
+
+class TestPil:
+
+    def to_img(self, x: Any) -> PIL.Image:
+        return x
+
+    def test_pil(self, img_tbl: Table) -> None:
```
Comment (mkornacker) [inline pixeltable/tests/functions/test_pil.py:9]: leave TODO: we should compare the actual outputs.

**Review by mkornacker — APPROVED** [review]

**Pull request merged**